core/drivers/crypto/caam/caam_jr.c

```diff
@@ -0,0 +1,676 @@
+
+	do {
+		/* Call the do_jr_dequeue function to dequeue the jobs */
+		job_complete = do_jr_dequeue(job_ids);
+
+		if (job_complete & job_ids)
+			return CAAM_NO_ERROR;
+
+		/* Check if JR interrupt otherwise wait a bit */
+		if (!caam_hal_jr_check_ack_itr(jr_privdata->baseaddr))
+			caam_udelay(10);
+	} while (infinite || (nb_loop--));
+
+	return CAAM_TIMEOUT;
+}
+
+enum caam_status caam_jr_enqueue(struct caam_jobctx *jobctx, uint32_t *job_id)
+{
+	enum caam_status retstatus = CAAM_FAILURE;
+	__maybe_unused int timeout  = 10; /* Nb loops to pool job completion */
+
+	if (!jobctx)
+		return CAAM_BAD_PARAM;
+
+	JR_DUMPDESC(jobctx->desc);
+
+	if (!jobctx->callback && job_id) {
+		JR_TRACE("Job Callback not defined whereas asynchronous");
+		return CAAM_BAD_PARAM;
+	}
+
+	if (jobctx->callback && !job_id) {
+		JR_TRACE("Job Id not defined whereas asynchronous");
+		return CAAM_BAD_PARAM;
+	}
+
+	jobctx->completion = false;
+	jobctx->status = 0;
+
+	/*
+	 * If parameter job_id is NULL, the job is synchronous, hence use
+	 * the local job_done callback function
+	 */
+	if (!jobctx->callback && !job_id) {
+		jobctx->callback = job_done;
+		jobctx->context = jobctx;
+	}
+
+	retstatus = do_jr_enqueue(jobctx, &jobctx->id);
+
+	if (retstatus != CAAM_NO_ERROR) {
+		JR_TRACE("enqueue job error 0x%08" PRIx32, retstatus);
+		return retstatus;
+	}
+
+	/*
+	 * If parameter job_id is defined, the job is asynchronous, so
+	 * returns with setting the job_id value
+	 */
+	if (job_id) {
+		*job_id = jobctx->id;
+		return CAAM_PENDING;
+	}
+
+#ifdef TIMEOUT_COMPLETION
+	/*
+	 * Job is synchronous wait until job completion or timeout
+	 */
+	while (!jobctx->completion && timeout--)
+		caam_jr_dequeue(jobctx->id, 100);
+
+	if (timeout <= 0) {
+		/* Job timeout, cancel it and return in error */
+		caam_jr_cancel(jobctx->id);
+		retstatus = CAAM_TIMEOUT;
+	} else {
+		if (JRSTA_SRC_GET(jobctx->status) != JRSTA_SRC(NONE))
+			retstatus = CAAM_JOB_STATUS;
+		else
+			retstatus = CAAM_NO_ERROR;
+	}
+#else
+	/*
+	 * Job is synchronous wait until job complete
+	 * Don't use a timeout because there is no HW timer and
+	 * so the timeout is not precise
+	 */
+	while (!jobctx->completion)
+		caam_jr_dequeue(jobctx->id, 100);
+
+	if (JRSTA_SRC_GET(jobctx->status) != JRSTA_SRC(NONE))
+		retstatus = CAAM_JOB_STATUS;
+	else
+		retstatus = CAAM_NO_ERROR;
+#endif
+
+	/* Erase local callback function */
+	jobctx->callback = NULL;
+
+	return retstatus;
+}
+
+enum caam_status caam_jr_init(struct caam_jrcfg *jrcfg)
+{
+	enum caam_status retstatus = CAAM_FAILURE;
+
+	JR_TRACE("Initialization");
+
+	/* Allocate the Job Ring resources */
+	retstatus = do_jr_alloc(&jr_privdata, jrcfg->nb_jobs);
+	if (retstatus != CAAM_NO_ERROR)
+		goto end_init;
+
+	jr_privdata->ctrladdr = jrcfg->base;
+	jr_privdata->jroffset = jrcfg->offset;
+
+	retstatus =
+		caam_hal_jr_setowner(jrcfg->base, jrcfg->offset, JROWN_ARM_S);
+	JR_TRACE("JR setowner returned 0x%" PRIx32, retstatus);
```

Comment:
s/`%08" PRIx32`/`%08x"`/  
(`retstatus` is an enum and will evaluate as an `int` or `unsigned int`, not explicitly a 32bit value)

Reply:
ok